File: ios/MyCustomPackage/MyCustomModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MyCustomModule, NSObject)


RCT_EXTERN_METHOD(openNativeScreen:(NSString *)message)

// Required for DispatchQueue.main.async
+ (BOOL)requiresMainQueueSetup
{
    return YES;
}

@end

File: ios/MyCustomPackage/MyCustomModule.swift
@objc (MyCustomModule)
class MyCustomModule: NSObject {
  
  @objc(openNativeScreen:)
  func openNativeScreen(message: NSString) -> Void {
    print(message);
    DispatchQueue.main.async {
      let appDelegate = UIApplication.shared.delegate as! AppDelegate
      let viewController = NativeViewController()
      viewController.message = message as String
      viewController.modalPresentationStyle = .fullScreen
      appDelegate.window?.rootViewController?.present(viewController, animated: false)
    }
  }
}





